### DISCUSSION TITLE
Git version check reveals environments with insufficient git version

### DISCUSSION THREAD

**Discussion by syphernl** [renovatebot/renovate#13026] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

29.36.2

### Please select which platform you are using if self-hosting.

gitlab.com

## Issue
Renovate mentions that our Git version is too old:

```txt
ERROR: Git version needs upgrading
       "detectedVersion": "2.32.0",
       "minimumVersion": "2.33.0"
FATAL: git version needs upgrading
```

We normally run Renovate within a pipeline with a container based on the `node:14.18.2-alpine3.14` base container. 
This particular container  contains Git `2.32.0`. 

Alpine 3.15 has a version that matches (`2.34.1`) but Node [does not have containers based on Alpine 3.15](https://hub.docker.com/_/node?tab=tags&page=1&name=alpine3.15).
Testing this on my workstation (Ubuntu 21.10) also fails as it contains Git version `2.32.0`, which is the most up-to-date version in the repository currently.

It seems that this specific version requirement could be problematic on a lot of platforms.

## Our (temporary) workaround
For now we worked around this by adding the following to `/etc/apk/repositories`:
```
@upstream http://nl.alpinelinux.org/alpine/v3.15/main
```

And installing the Git version from this particular repo:

```
apk add git@upstream
```

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste any log here, between the starting and ending backticks
```

</details>

### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by viceice** [comment, marked as answer]
That git version is required since v27 of renovate and is now enforced.

Use our official `renovate/renovate` or `renovate/node` images.

**Reply by syphernl** [reply]
Thanks, I was not aware of this. I still believe this requirement might be a little too strict, as not many platforms contain Git >= 2.33.0 at this time.

For now we'll keep on using our workaround since the official containers are too big in comparison to those based on Alpine. 
The `renovate/renovate` container is 1.25GB while our custom renovate container is 185.43 MiB.

**Reply by viceice** [reply]
what about the `renovate/renovate:slim`  or `renovate/node:14` images?

```sh
> docker image ls renovate/*
REPOSITORY          TAG       IMAGE ID       CREATED        SIZE
renovate/node       14        88b4a24c280b   4 hours ago    279MB
renovate/renovate   latest    941b57759e42   6 hours ago    3.54GB
renovate/renovate   slim      d938be4a5610   7 hours ago    477MB
renovate/renovate   <none>    358f679ffa46   25 hours ago   3.54GB
renovate/node       16        bef36689453e   6 days ago     267MB
renovate/node       latest    8be2f626ca7a   2 weeks ago    279MB
renovate/node       <none>    0512288471a0   8 weeks ago    280MB
renovate/renovate   <none>    798de25977af   2 months ago   422MB
```

I've also an unofficial copy of the slim image at https://github.com/visualon/docker-images/pkgs/container/renovate to prevent docker hub limits 😉

**Reply by viceice** [reply]
You should read the changes when we do renovate major bumps: https://github.com/renovatebot/renovate/releases/tag/27.0.0
![image](https://user-images.githubusercontent.com/1798109/145368159-212c2770-1f04-48a4-9ae1-2be98fa0089d.png)

**Reply by syphernl** [reply]
> what about the `renovate/renovate:slim` or `renovate/node:14` images?

Thanks, I'll have a look at them
